src/Pinch/Internal/Value.hs

```diff
@@ -70,6 +70,7 @@ data Value a where
 
     VMap  :: forall k v. (IsTType k, IsTType v)
           => !(FoldList (MapItem k v)) -> Value TMap
+    VNullMap :: Value TMap
```

Comment:
It might be worth adding a test that round trips an actual empty map (not a `Value`) to compact and back.